DFC.EventGridSubscriptions.ApiFunction/Function/Execute.cs

```diff
@@ -0,0 +1,179 @@
+﻿using DFC.EventGridSubscriptions.Data;
+using DFC.EventGridSubscriptions.Data.Models;
+using DFC.EventGridSubscriptions.Services.Interface;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using System.Web.Http;
+
+namespace DFC.EventGridSubscriptions.ApiFunction
+{
+    /// <summary>
+    /// The Execute Function.
+    /// </summary>
+    public class Execute
+    {
+        private readonly ISubscriptionRegistrationService subscriptionRegistrationService;
+        private readonly IOptionsMonitor<AdvancedFilterOptions> advancedFilterOptions;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Execute"/> class.
+        /// </summary>
+        /// <param name="subscriptionRegistrationService">A Subscription Registration Service.</param>
+        /// <param name="advancedFilterOptions">The Advanced Filter Options.</param>
+        public Execute(ISubscriptionRegistrationService subscriptionRegistrationService, IOptionsMonitor<AdvancedFilterOptions> advancedFilterOptions)
+        {
+            this.subscriptionRegistrationService = subscriptionRegistrationService;
+            this.advancedFilterOptions = advancedFilterOptions;
+        }
+
+        /// <summary>
+        /// Runs the function.
+        /// </summary>
+        /// <param name="req">The Request.</param>
+        /// <param name="log">The Logger.</param>
+        /// <param name="subscriptionName">The subscription name.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        [FunctionName("Execute")]
+        public async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "post", "delete", Route = "Execute/{subscriptionName?}")] HttpRequest req, ILogger log, string subscriptionName)
+        {
+            try
+            {
+                log.LogInformation("Subscription function execution started");
+
+                if (req == null || string.IsNullOrEmpty(req.Method))
+                {
+                    throw new ArgumentNullException(nameof(req));
+                }
+
+                return req.Method.ToUpperInvariant() switch
+                {
+                    "POST" => await HandlePostAsync(req, log).ConfigureAwait(false),
```

Comment:
The `ConfigureAwait(false)`'s aren't required in (.net core) Azure Functions, as there's no `SynchronizationContext`.
See [1](https://devblogs.microsoft.com/dotnet/configureawait-faq/), [2](https://stackoverflow.com/questions/55542016/is-configureawaitfalse-needed-beneficial-when-awaiting-async-calls-in-azure-fu).

Reply:
The joys of SonarCloud :(